lib/foreman/renderer/scope/macros/host_template.rb

```diff
@@ -121,6 +121,39 @@ def ks_console
             (@port && @baud) ? "console=ttyS#{@port},#{@baud}" : ''
           end
 
+          apipie :method, 'Generates commands for package(s) installation, supported OS families: Redhat, Debian & Suse' do
+            required :packages, String, desc: "Package(s) to install"
+            returns String, desc: 'Installation commands'
+            example "CentOS host: install_packages('pkg1') #=> yum -y install pkg1"
+            example "Fedora host: install_packages('pkg1 pkg2') #=> dnf -y install pkg1 pkg2"
+          end
+          def install_packages(packages)
+            raise UndefinedPackages.new if packages.empty?
+
+            case host.operatingsystem&.family
+            when 'Redhat'
+              if host.operatingsystem.name.downcase == 'fedora'
+                "dnf -y install #{packages}"
+              else
+                "yum -y install #{packages}"
+              end
+            when 'Debian'
+              <<~CMD
+                [ -x "$(command -v subscription-manager)" ] && subscription-manager refresh
```

Comment:
This is interesting. I never considered using `-x` on output of `command -v` but it does look like it returns non-executable files in `$PATH`. I learned something.

One thing I would consider here. If it runs in bash with `set -e` then I think it will exit if subscription-manager is not installed. If it's a real `if [ -x ...] ; then .. fi` block then you don't have this problem.

Reply:
I guess this was added because subscription-manager can live in /usr/sbin/subscription-manager vs /usr/bin/subscription-manager

I think kickstart `set -e` did not work and registration does not use it, if it's a real issue, user would face it with rex templates. Given this was added in rex a [long time ago](https://github.com/theforeman/community-templates/commit/aeea6bca498e735894823f11bd14536997f6eb0c), I don't think it's broken and most likely we don't have to change it here.

Reply:
> I guess this was added because subscription-manager can live in /usr/sbin/subscription-manager vs /usr/bin/subscription-manager

Not really. Normally `command -v` behaves like `which` but without printing to STDERR if it isn't found. It does also find shell built ins which `which` doesn't However, it looks like it can also find files in `$PATH` if they aren't executable. A bit of an edge case but a valid check.

Reply:
I tried it and it won't exit if `subman` is not present:

```sh
#!/bin/sh

set -e
[ -x "$(command -v subscription-manager)" ] && subscription-manager refresh
echo "Say hi!"

# => Say hi!
```

Reply:
Interesting, I didn't expect `set -e` to behave that way but it does look like it continues even with exit code 1. Looks like it continues regardless of anything when `&&` is used:
```console
$ cat bla
#!/bin/bash
set -e
[ -x "$(command -v false)" ] && false
echo $?
$ ./bla
1
$ cat bla2
#!/bin/bash
set -e
[ -x "$(command -v dfsfdfgdf)" ]
echo $?
$ ./bla2
$
```

So that means with `set -e`, `subscription-manager refresh` can fail and the script would still continue.